app/queries/developer_query.rb

```diff
@@ -0,0 +1,81 @@
+class DeveloperQuery
+  include Pagy::Backend
+
+  alias_method :build_pagy, :pagy
+
+  attr_reader :options
+
+  def initialize(options = {})
+    @options = options
+  end
+
+  def available?
+    return nil if options[:available] == ""
+
+    options[:available]
+  end
+
+  def search_field
+    return nil if options[:search_field] == ""
+
+    options[:search_field]
+  end
+
+  def search_status
+    return nil if options[:search_status] == ""
+
+    options[:search_status]
+  end
+
+  def pagy
+    @pagy ||= initialize_pagy.first
+  end
+
+  def records
+    @records ||= initialize_pagy.last
+  end
+
+  private
+
+  def initialize_pagy
+    records = Developer.most_recently_added.with_attached_avatar
+
+    if available? && search_field && search_status
+      records = records.where("technical_skills LIKE ? OR pivot_skills LIKE ? AND search_status = ?", "%#{search_field}%", "%#{search_field}%", "%#{search_status}%")
+      records = records.available
+    end
+
+    if available? && search_field
+      records = records.where("technical_skills LIKE ? OR pivot_skills LIKE ?", "%#{search_field}%", "%#{search_field}%")
+      records = records.available
+    end
+
+    if available? && search_status
+      records = records.where("search_status = ?", "%#{search_status}%")
```

Comment:
```suggestion
      records = records.where(search_status: search_status)
```

Should help prevent SQL injection.